src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/ApplicationController.cs

```diff
@@ -287,6 +302,17 @@ public async Task<IActionResult> Edit(string id, string returnUrl = null)
                 _notifier.Warning(H["There are no registered services to provide roles."]);
             }
 
+            var scopes = await _applicationManager.GetPermissionsAsync(application);
```

Comment:
`permissions` may be a better name here. Alternatively, you can also use the `_applicationManager.HasPermission(...)` method in the foreach body.

Reply:
Will rename